src/app/src/main/kotlin/org/imperial/mrc/hint/Extensions.kt

```diff
@@ -66,7 +67,7 @@ fun Response.asResponseEntity(): ResponseEntity<String> {
 
 }
 
-fun Request.getStreamingResponseEntity(): ResponseEntity<StreamingResponseBody> {
+fun Request.getStreamingResponseEntity(headRequest: (url: String, parameters: Parameters?) -> Request): ResponseEntity<StreamingResponseBody> {
```

Comment:
This function parameter for fetching the head response is a bit weird, but makes it much easier to unit test..

Reply:
Ok, yeah it's pretty weird but I can see the issue with testing.